File: Diff_wepsites_questions/Two_sum.cpp
#include<iostream>
using namespace std;


int TwoSum(int arr[],int n,int target){
    for(int i = 0; i<n; i++){
        for(int j =i+1; j<n; j++){
          if(arr[i]+arr[j]==target){
            cout<< i<<" "<<j<<endl;
          }

        }
    }
}

int main(){
int arr [4]={2,7,11,15};
int targert = 9;
TwoSum(arr,4,9);

    return 0;
}

File: Neet_Code/Two_Sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        
      Map<Integer,Integer> mp = new HashMap<>();
      
        for(int i = 0; i<nums.length;i++){
            
            int val  = target - nums[i];
        
            if(mp.containsKey(val)){
                
                return new int []{mp.get(val),i};
            }
            
            mp.put(nums[i],i);
        }
        return new int []{};
    }
}
